service/frontend/workflow_handler.go

```diff
@@ -1815,6 +1908,11 @@ func (wh *WorkflowHandler) RequestCancelWorkflowExecution(ctx context.Context, r
 		return nil, errRequestNotSet
 	}
 
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(
+		telemetry.WorkflowIDKey(request.WorkflowExecution.WorkflowId),
```

Comment:
For the record note, this fixes a bug an the original OTEL instrumentation commit: https://github.com/stephanos/temporal/blob/bea77272f0b4f7a01fba897c5a2e39f1563015f2/service/frontend/workflow_handler.go#L1899